File: onlinejudge/leetcode/9-Palindrome Number.java
class Solution {
    public boolean isPalindrome(int x) {
        if (x < 0) return false;

        var charArray = String.valueOf(x).toCharArray();
        for (int i = 0; i < charArray.length / 2; i += 1) {
            if (charArray[i] != charArray[charArray.length - 1 - i]) {
                return false;
            }
        }
        return true;
    }
}

File: onlinejudge/leetcode/9-Palindrome Number.go
func isPalindrome(x int) bool {
	if x < 0 {
		return false
	}

	str := strconv.Itoa(x)
	for i := 0; i < len(str) / 2; i += 1 {
		if str[i] != str[len(str)-1-i] {
			return false
		}
	}
	return true
}
